### PR TITLE
arch: Review get_gpu_info

### PR DESCRIPTION
**Pull request by Leitevmd**
@FabioLuporini , please check review comments. We could also improve another points (e.g.,  get physical id also from `lspci` and `nvidia-smi`)

### CODE REVIEW

File: devito/arch/archinfo.py [hunk 1]
```diff
@@ -239,7 +245,7 @@ def nvidiasmi_gpu_info(text):
     def homogenise_gpus(gpu_infos):
         if gpu_infos == []:
             warning('No graphics cards detected')
-            return None
+            return {}
```
Comment (Leitevmd) [inline devito/arch/archinfo.py:248]: @FabioLuporini this is what you requested

File: devito/arch/archinfo.py [hunk 2]
```diff
@@ -251,7 +257,7 @@ def homogenise_gpus(gpu_infos):
 
         warning('Different models of graphics cards detected')
 
-        return None
+        return {}
```
Comment (Leitevmd) [inline devito/arch/archinfo.py:260]: Also here

File: devito/arch/archinfo.py [hunk 3]
```diff
@@ -227,8 +227,14 @@ def nvidiasmi_gpu_info(text):
                 gpu_info = {}
                 match = re.match(r'GPU *[0-9]*\: ([\w]*) (.*) \(', line)
                 if match:
-                    gpu_info['architecture'] = match.group(1)
-                    gpu_info['product'] = match.group(2)
+                    if match.group(1) == 'Graphics':
```
Comment (Leitevmd) [inline devito/arch/archinfo.py:230]: @FabioLuporini This is for the output you showed me on slack.. let me know what you think

File: tests/test_gpu_common.py [hunk 4]
```diff
@@ -19,9 +19,10 @@ def test_get_gpu_info(self):
         info = get_gpu_info()
         try:
             assert info['architecture'].lower() in ['tesla', 'geforce', 'unspecified']
-        except TypeError:
+        except KeyError:
```
Comment (Leitevmd) [inline tests/test_gpu_common.py:22]: Here we need to define what we want.. let pass only these values or even empty info (which was previous behavior)

Comment (FabioLuporini) [reply tests/test_gpu_common.py:22]: yeah I think the current behaviour is OK

File: tests/test_gpu_common.py [hunk 5]
```diff
@@ -19,9 +19,10 @@ def test_get_gpu_info(self):
...
             # There might be than one GPUs, but for now we don't care
             # as we're not really exploiting this info yet...
+            print("KeyError")
```
Comment (FabioLuporini) [inline tests/test_gpu_common.py:25]: leftover, drop?

File: devito/arch/archinfo.py (hunk 3 above)
Comment (FabioLuporini) [reply devito/arch/archinfo.py:230]: Looks good to me

**Comment by FabioLuporini** [comment]
Thanks, merging

**Pull request merged**